### PR TITLE
Attributes for devices

### PR DESCRIPTION
**Pull request by krwq**
Read README.md for description.

Contributes to: #567, #418

cc: @pgrawehr @colombod

### CODE REVIEW

File: src/devices/System.Device.Model/README.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+## Type serialization
+
+Only simple types can be serialized:
+- enums (without Flags attribute)
+  - values out of enum range are not permitted (i.e. bitwise combination)
+- UnitsNet units
+- basic C# types
+- Vector2, Vector3, Vector4
+- Color
```
Comment (colombod) [inline src/devices/System.Device.Model/README.md:111]: Maybe this should be fully qualified type names to make it easier to user what we refer to

Comment (colombod) [reply src/devices/System.Device.Model/README.md:111]: Should we do the switch to portable library like ImageSharp at this point?

Comment (krwq) [inline src/devices/System.Device.Model/README.md:111]:
```suggestion
- System.Numerics.Vector2, System.Numerics.Vector3, System.Numerics.Vector4
- System.Drawing.Color
```

**Review by colombod — APPROVED** [review]

File: src/devices/System.Device.Model/README.md (hunk 1 above)
Comment (krwq) [reply src/devices/System.Device.Model/README.md]: this doesn't really depend on it yet, once we convert Iot.Device.Bindings to ImageSharp that's when we should deprecate serialization of Color, that's why it's still experimental. Also only serializer will really depend on System.Drawing not the library itself

File: src/devices/Ina219/Ina219.cs [hunk 2]
```diff
@@ -6,20 +6,22 @@
...
-    /// Binding that exposes an INA219 Bidirectional Current/Power monitor.
+    /// INA219 Bidirectional Current/Power monitor.
     ///
     /// The INA219 is a high-side current shunt and power monitor with an I2C interface.
     /// The INA219 monitors both shunt drop and supply voltage, with programmable conversion
     /// times and filtering. A programmable calibration value, combined with an internal multiplier,
     /// enables direct readouts in amperes. An additional multiplying register calculates power in watts.
     /// <see href="http://www.ti.com/lit/ds/symlink/ina219.pdf"/>
     /// </summary>
+    [Interface("INA219 Bidirectional Current/Power monitor")]
```
Comment (Ellerbach) [inline src/devices/Ina219/Ina219.cs:24]: are there any special character not allowed or which needs to be escaped? in this case the `/` ?

File: src/devices/SenseHat/SenseHat.cs [hunk 3]
```diff
@@ -53,96 +83,96 @@ public SenseHat(I2cBus? i2cBus = null, bool shouldDispose = false)
         /// Fills LED matrix with specified color
         /// </summary>
         /// <param name="color">Color to fill LED matrix with</param>
-        public void Fill(Color color = default(Color)) => _ledMatrix.Fill(color);
+        public void Fill(Color color = default(Color)) => LedMatrix.Fill(color);
```
Comment (Ellerbach) [inline src/devices/SenseHat/SenseHat.cs:86]: Can those type of functions be `[Command]` ?

File: src/devices/System.Device.Model/README.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+For properties providing name of the `Telemetry` is optional as it can be deduced from the property name.
+
+Telemetry can be put on:
+- properties
+- methods returning value but not taking any arguments
+- methods returning bool and taking one `out` argument
+  - multiple `out` arguments are currently out of scope but are considered
+
+```csharp
+        [Telemetry]
```
Comment (Ellerbach) [inline src/devices/System.Device.Model/README.md:40]: remove the extra space, that's not needed, can do everywhere in the samples from this readme

**Review by Ellerbach — APPROVED** [review]
Few comments, great work! Looking forward for the next steps!

File: src/devices/SenseHat/SenseHat.cs (hunk 3 above)
Comment (krwq) [reply src/devices/SenseHat/SenseHat.cs:86]: I wasn't sure about annotating any of the displays yet so I left them out of scope for now.

File: src/devices/Ags01db/Ags01db.cs [hunk 5]
```diff
@@ -23,18 +31,14 @@ public class Ags01db : IDisposable
...
         /// ASG01DB VOC (Volatile Organic Compounds) Gas Concentration (ppm)
         /// </summary>
-        public double Concentration => GetConcentration();
+        [Telemetry]
+        public Ratio Concentration => Ratio.FromPartsPerMillion(GetConcentration());
 
         /// <summary>
         /// ASG01DB Version
         /// </summary>
         public byte Version => GetVersion();
```
[the remark this answers is not in the document]
Comment (krwq) [reply src/devices/Ags01db/Ags01db.cs:41]: I might have missed some on first couple of sensors when I still wasn't sure if these should be properties, I've fixed couple of more cases.

File: src/devices/SenseHat/SenseHat.cs (hunk 3 above)
Comment (Ellerbach) [reply src/devices/SenseHat/SenseHat.cs:86]: Fair, let's see first with the first sensors. Impatient to see all this coming together :-)

**Pull request merged**